src/main/java/com/puppycrawl/tools/checkstyle/checks/TranslationCheck.java

```diff
@@ -388,7 +388,7 @@ private void checkExistenceOfDefaultTranslation(ResourceBundle bundle) {
      *
      * @param bundle resource bundle.
      */
-    @SuppressWarnings("determinism:argument.type.incompatible")  // true positive; logging: Iteration over OrderNonDet collection
+    @SuppressWarnings("determinism:argument.type.incompatible")  // false positive; logging: Iteration over OrderNonDet collection; setRequiredTranslations ensures files are stored in the order they are input
```

Comment:
I think it would be more appropriate to annotate `requiredTranslations` as `@Det`, because that is accurate.  (Suppress the warning at the initialization because it is never side-effected, only re-set.  Or, change the HashSet, which would eliminate the warning at the initialization.)